cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt

```diff
@@ -80,6 +80,158 @@ class StatementHandler(frontend: PythonLanguageFrontend) :
         }
     }
 
+    /**
+     * Translates a Python [`With`](https://docs.python.org/3/library/ast.html#ast.With) into a
+     * [Block].
+     *
+     * We return a Block to handle the with statement, following
+     * [python's documentation](https://docs.python.org/3/reference/compound_stmts.html#the-with-statement).
+     * The context manager's `__enter__` and `__exit__` methods should be identified before entering
+     * the try-block. However, we simplify the code from the documentation as follows:
+     * * `__enter__()` is called before the try-block.We make the identification and the call in the
+     *   same step.
+     * * `__exit__()` is either called in the `except`-block or in the `finally`-block but not
+     *   identified separately.
+     * * In fact, the `finally` is used like an `else`-block, so we use this construction.
+     *
+     * Example: We will translate the code
+     *
+     * ```python
+     * with ContextManager() as cm:
+     *     cm.doSomething()
+     * ```
+     *
+     * to something like
+     *
+     * ```python
+     * manager = ContextManager()
+     * tmpVal = manager.__enter__()
+     * try:
+     *     cm = tmpVal # Doesn't exist if no variable is used
+     *     cm.doSomething()
+     * except:
+     *     if not manager.__exit__(*sys.exc_info()):
+     *         raise
+     * else:
+     *     manager.__exit__(None, None, None)
+     * ```
+     */
+    private fun handleWithStatement(node: Python.AST.With): Block {
```

Comment:
This function is quiet a hand full. I think it might be easier to read, if we split it into multiple helpers. for creating all the required implicit cpg nodes. I'll create a PR to see if it's actually better...

Reply:
Since adding the multi-resource thing required a heavy refactoring, it should already be simplified a bit

Reply:
Looks pretty good now, thanks :)